Lesson 23

AndreyPushai opened the pull request:
Added completed homework.

AndreyPushai committed:
Upd gitignore

Fixed typo

AndreyPushai committed:
Upd package.json

- Added dotenv dependency
- upd script args

AndreyPushai committed:
Upd playwright.config.ts

- added dotenv configs
- upd reporter
- added baseURL and httpCredentials

AndreyPushai committed:
Added .env file

AndreyPushai committed:
Added PageObject models

- reworked from cypress project

AndreyPushai committed:
Added registration.spec.ts

- and removed example.spec.ts
- finished homework task.

apolonskiy commented on poms/BasePage.ts:
```diff
@@ -0,0 +1,8 @@
+    /**
+     * @param {import('@playwright/test').Page} page
+     */
+    constructor(page){
```
У вас же ТС, нащо вам ЖС док, робіть тайп пейджі))

AndreyPushai replied:
Виправив.

apolonskiy commented on poms/RemoveAccountPopUp.ts:
```diff
@@ -0,0 +1,6 @@
+
+export default class RemoveAccountPopUp extends BasePopUp {
+    public popUp = this.page.locator("app-remove-account-modal");
+    public removeButton = this.popUp.locator("button", {hasText: "Remove"});
```
Можливо метод з підтвердженням ремуву?

AndreyPushai replied:
Done.

apolonskiy commented on poms/SettingsPage.ts:
```diff
@@ -0,0 +1,5 @@
+import BasePage from "./BasePage";
+
+export default class SettingsPage extends BasePage {
+    public removeMyAccountButton = this.page.locator("button", {hasText: "Remove my account"});
```
І тут логічно описати що саме робити на сторінці з кнопкою

AndreyPushai replied:
Done.

apolonskiy commented on poms/Sidebar.ts:
```diff
@@ -0,0 +1,11 @@
+import BasePage from "./BasePage";
+
+export default class Sidebar extends BasePage{
```
Вони всі посилання, можна зробити один універсальний текстовий селектор і один метод, т.я. у вас Тісто можна одразу в типів аргументу назви рабів передати і через hasText вибрати потрібний)

AndreyPushai replied:
Done.

AndreyPushai committed:
Upd BasePage.ts

- Fixed typing in TS convention

AndreyPushai committed:
Upd BasePopUp.ts

- added verification methods

AndreyPushai committed:
Upd LoginPopUp.ts

- fixed fillAndSubmitForm return type

AndreyPushai committed:
Upd RemoveAccountPopUp.ts

- added confirmRemoveAccount method

AndreyPushai committed:
Upd SettingsPage.ts

- Added clickRemoveMyAccountButton method

AndreyPushai committed:
Upd Sidebar.ts

- added SidebarLinks type
- Upd POM class, replaced links with getLink method
- added clickLink method

AndreyPushai committed:
Upd registration.spec.ts

- fixed imports
- upd positive test with fixes and updated POMs
